radicle-node/src/tests.rs

```diff
@@ -600,6 +600,61 @@ fn test_refs_announcement_relay() {
     );
 }
 
+#[test]
+fn test_refs_announcement_fetch() {
+    logger::init(log::Level::Debug);
+
+    let tmp = tempfile::tempdir().unwrap();
+    let mut alice = Peer::config(
+        "alice",
+        [7, 7, 7, 7],
+        Storage::open(tmp.path().join("alice")).unwrap(),
+        peer::Config::default(),
+    );
+    let eve = Peer::config(
+        "eve",
+        [8, 8, 8, 8],
+        Storage::open(tmp.path().join("eve")).unwrap(),
+        peer::Config::default(),
+    );
+
+    let bob = {
+        let mut rng = fastrand::Rng::new();
+        let signer = MockSigner::new(&mut rng);
+        let storage = fixtures::storage(tmp.path().join("bob"), &signer).unwrap();
```

Comment:
There shouldn't be a need to create real storages for any of the peers, unless you need these peers to actually have the inventory?

You can simply generate an rid with `arbitrary::gen::<Id>(1)` and use that for bob's `ref_announcement` call. Alice doesn't need to know that bob actually doesn't have the inventory.

This would simplify the test a lot as you can just use `Peer::new`

Reply:
The path I wanted to test is:  

`alice` already has `repo A` in its storage.  Then `alice` gets `bob`'s `refs_ann`, and `alice` should not fetch because `bob` is not trusted yet. 

I was not able to create such case in this test, unless I follow the same idea I used in `e2e.rs`.  Are you saying I should be able to mimic such case without creating actual storages and repos?